src/UIObject.cpp

```diff
@@ -0,0 +1,20 @@
+#include "UIObject.h"
+
+#include <glm/gtc/matrix_transform.hpp>
+
+UIObject::UIObject(glm::vec2 origin, glm::vec3 colour, glm::vec2 size, MeshAsset* mesh, TextureAsset* tex, ShaderAsset* shader)
+    : origin_(origin)
+    , colour_(colour)
```

Comment:
Doesn't the shader dictate the colour?

Reply:
In the future, we probably want to pass textures to the UIObjects but since that isn't working right now and is not a priority, we just gave it a colour. This will be phased out once we get to that part of development.